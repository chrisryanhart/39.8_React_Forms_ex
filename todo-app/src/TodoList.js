import React, {useState} from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import { v4 as uuid } from "uuid";

function TodoList() {
    
    const INITIAL_STATE = [{id: uuid(), task:"Cut the grass"},{id: uuid(),task:"Pull weeds"}];
    const [list,setList] = useState(INITIAL_STATE);

    const remove = (id) => {

        setList(list.filter(ele => ele.id !== id));
    }

    const addTodo = (text) => {
        setList(data => [
            ...data,
            {id: uuid(), task:text}
        ])
    }

    const todos = list.map(ele=> <Todo id={ele.id} key={ele.id} remove={remove} text={ele.task}/>);


    return (
      <div className="App">
        <h2>Todolist</h2>
        {todos}

        <NewTodoForm addTodo={addTodo}/>
      </div>
    );
}

export default TodoList;