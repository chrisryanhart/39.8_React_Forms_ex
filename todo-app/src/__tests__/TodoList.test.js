import React from 'react';
import { fireEvent, render } from "@testing-library/react";
import TodoList from '../TodoList';
// import '@testing-library/jest-dom';


it("renders without crashing", function() {
    render(<TodoList/>);
})

it("renders the same output", function() {
    const {asFragment} = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
})

it("can add a new todo", function() {
    const { getByLabelText, queryByText, debug } = render(<TodoList/>);

    expect(queryByText('-Pay Taxes')).not.toBeInTheDocument();

    const textInput = getByLabelText('New Todo:');
    const submitBtn = queryByText('Add Todo');

    fireEvent.change(textInput, {target: {value: "Pay Taxes"}});
    fireEvent.click(submitBtn);

    expect(queryByText('-Pay Taxes')).toBeInTheDocument();
})

it("can delete a todo", function() {
    const { getByLabelText, queryByText, getAllByText } = render(<TodoList/>);

    expect(queryByText('-Cut the grass')).toBeInTheDocument();

    const deleteBtn = getAllByText('X')[0];

    fireEvent.click(deleteBtn);

    expect(queryByText('-Cut the grass')).not.toBeInTheDocument();
})