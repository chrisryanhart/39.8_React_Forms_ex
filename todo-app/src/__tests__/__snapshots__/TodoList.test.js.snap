// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the same output 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h2>
      Todolist
    </h2>
    <div>
      -Cut the grass
      <button>
        X
      </button>
    </div>
    <div>
      -Pull weeds
      <button>
        X
      </button>
    </div>
    <div>
      <form>
        <label>
          New Todo:
        </label>
        <input
          value=""
        />
        <button>
          Add Todo
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
